src/psyclone/psyGen.py

```diff
@@ -1661,8 +1668,33 @@ def __init__(self, KernFactory, BuiltInFactory, alg_calls=None):
         Schedule.__init__(self, sequence=sequence, parent=None)
         self._invoke = None
         self._opencl = False  # Whether or not to generate OpenCL
+        # InvokeSchedule opencl_options default values
+        self._opencl_options = {"end_barrier": True}
         self._name_space_manager = NameSpaceFactory().create()
 
+    def set_opencl_options(self, options):
+        '''
+        Validate and store a set of options associated to the InvokeSchedule to
+        tune the OpenCL code generation.
+
+        :param options: a set of options to tune the OpenCL code.
+        :type options: map of <string>:<value>
+
+        '''
+        # Validate that the options given are supported and store them
+        for key, value in options.items():
+            if key == "end_barrier":
+                if not isinstance(value, bool):
+                    raise TypeError(
+                        "InvokeSchedule opencl_option 'end_barrier' "
+                        "should be a boolean.")
+            else:
+                raise AttributeError(
+                    "InvokeSchedule does not support the opencl_option '{0}'."
```

Comment:
Done. I used the plural tense "Accepted options are: 'end_barrier'." as this may accept more options if the future. Is this right? or should I use the singular form for now?